overlord/snapstate/handlers.go

```diff
@@ -977,15 +977,20 @@ func (m *SnapManager) doLinkSnap(t *state.Task, _ *tomb.Tomb) error {
 	return nil
 }
 
+func snapdSnapInstalled(st *state.State) bool {
+	var snapst SnapState
+	err := Get(st, "snapd", &snapst)
+	return err == nil
```

Comment:
Should we also check something about `snapst` here? That there is one active revision or that there are any revisions at all? I'm worried that through manipulation of the state there will be a "yeah I'm here but I'm empty inside" entry.

Reply:
The original code in https://github.com/snapcore/snapd/pull/6492#discussion_r256292764 was doing that - I can change it again if that is desired.